src/pb_utils.h

```diff
@@ -165,18 +167,37 @@ struct ResponseBatch : SendMessageBase {
 
 enum LogLevel { INFO = 0, WARNING, ERROR, VERBOSE };
 
+enum MetricKind { COUNTER, GAUGE };
+
 struct LogSendMessage : SendMessageBase {
   bi::managed_external_buffer::handle_t filename;
   int32_t line;
   bi::managed_external_buffer::handle_t log_message;
   LogLevel level;
 };
 
-
 struct CleanupMessage : SendMessageBase {
   void* id;
 };
 
+enum MetricFamilyRequestKind { MetricFamilyNew, MetricFamilyDelete };
+
+enum MetricRequestKind {
+  MetricNew,
+  MetricDelete,
+  MetricValue,
+  MetricIncrement,
+  MetricSet
+};
+
+struct CustomMetricsMessage : SendMessageBase {
+  bi::managed_external_buffer::handle_t message;
+  bool has_error;
+  bool is_error_set;
+  bi::managed_external_buffer::handle_t error;
+  double value;
+};
+
```

Comment:
We should move these enums to `pb_custom_metric.cc` and `custom_metrics_registery.cc`.